packages/hash/frontend/src/shared/layout/layout-with-sidebar/account-page-list/utils.ts

```diff
@@ -0,0 +1,123 @@
+import { UniqueIdentifier } from "@dnd-kit/core";
+import { arrayMove } from "@dnd-kit/sortable";
+import { AccountPage } from "../../../../components/hooks/useAccountPages";
+
+export interface TreeItem {
+  page: AccountPage;
+  depth: number;
+}
+
+export const getTreeItemList = (
+  pagesList: AccountPage[],
+  parentId: string | null = null,
+  depth = 0,
+): TreeItem[] => {
+  const emptyList: TreeItem[] = [];
+
+  return pagesList
+    .filter((page) => page.parentPageEntityId === parentId)
+    .reduce((prev, page) => {
+      const children = getTreeItemList(pagesList, page.entityId, depth + 1);
+
+      const item = {
+        page,
+        depth,
+      } as TreeItem;
```

Comment:
Does this work? If not, why not? 

```suggestion
      const item: TreeItem = {
        page,
        depth,
      };
```

Reply:
It works. Changed in 28b80ac08c80c99b30c7a7bb1f68d1418db7a2d5